mempool/src/shared_mempool/network.rs

```diff
@@ -113,3 +119,46 @@ impl ApplicationNetworkSender<MempoolSyncMsg> for MempoolNetworkSender {
         unimplemented!()
     }
 }
+
+type MempoolMultiNetworkSender = MultiNetworkSender<MempoolSyncMsg, MempoolNetworkSender>;
+
+pub(crate) struct MempoolNetworkInterface {
+    peer_metadata_storage: Arc<PeerMetadataStorage>,
+    sender: MempoolMultiNetworkSender,
+    sync_states: LockingHashMap<PeerNetworkId, PeerSyncState>,
+}
+
+impl NetworkInterface<MempoolSyncMsg, MempoolMultiNetworkSender> for MempoolNetworkInterface {
+    type AppDataKey = PeerNetworkId;
```

Comment:
interesting, I hadn't seen this in the previous code (I guess I never followed up)-- do we envision a case where this won't be PeerNetworkId?

Reply:
So, right now in `HealthCheckInterface` I think it's not a `PeerNetworkId` or if it is it's just shimmed that way.  I can envision a case where we may want to key on something other than PeerNetworkId, but in the current usecases probably not.

Reply:
I don't suppose Rust allows for a default type...